### PR TITLE
fix(scheduler): support both long and short sha being passed in for builds

### PR DESCRIPTION
**Pull request by helgi**

### CODE REVIEW

File: rootfs/scheduler/__init__.py [hunk 1]
```diff
@@ -601,11 +601,14 @@ def _create_rc(self, name, image, command, **kwargs):
...
         imgurl = self.registry + "/" + image
         TEMPLATE = RCD_TEMPLATE
-        shalen = len(image[image.index(":")+5:])
-        git = image[image.index(":")+1:image.index(":")+4]
-        if git == "git" and shalen == 8 and app_type == 'web':
-            imgurl = "http://"+settings.S3EP+"/git/home/"+image+"/push/slug.tgz"
+
+        # Check if it is a slug builder image.
+        # Example format: golden-earrings:git-5450cbcdaaf9afe6fadd219c94ac9c449bd62413s
+        vcs, sha = image[image.index(':')+1:].split('-')
```
Comment (arschles) [inline rootfs/scheduler/__init__.py]: @helgi what do you think about the following changes to the API:

- for buildpack builds, builder sends `quay.io/deis/slugrunner` (or other appropriate image as appropriate) as the image, and the URL of the slug as another param?
- for dockerfile builds (not yet implemented), builder sends the URL of the image (which would be in the internal registry) and no URL for the slug (since it's N/A). It would no longer send `"dockerfile": true`

Comment (helgi) [inline rootfs/scheduler/__init__.py]: yeah, we can do it as part of another PR. Create a ticket. That should take care of a few hardcoded points we have in the system as it stands

**Comment by smothiki** [comment]
I'm working on this issue https://github.com/deis/workflow/issues/141. which might completely avoid that condition

File: rootfs/scheduler/__init__.py (hunk 1 above)
Comment (arschles) [inline rootfs/scheduler/__init__.py]: sounds good @helgi

Comment (arschles) [inline rootfs/scheduler/__init__.py]: @helgi @smothiki I copied these suggestions to #141

**Comment by smothiki** [comment]
I have no worries about merging this PR but soon this code will get obsolete
